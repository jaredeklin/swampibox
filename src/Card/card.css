.card {
  border: 3px solid white;
  color: white;
  border-radius: 10px;
  width: 300px;
  height: 490px;
  display: inline-block;
  vertical-align: top;
  margin: 15px;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  box-sizing: border-box;
  padding: 0 20px;
  background-color: rgba(0, 0, 0, .8);
}

.card .active:hover {
  box-shadow: 0px 0px 30px 0px #FFFF00;
}

.card ul {
  padding-left: 0;
}

.card li {
  list-style: none;
  margin: 15px 0;
  font-size: 20px;
}

.card-keys {
  font-weight: bold;
  text-transform: uppercase;
}

.card h2 {
  font-size: 30px;
  margin-bottom: 10px;
  border-bottom: 3px solid #FFFF00;
  color: #FFFF00;
}

.card-button {
  height: 75px;
  background-color: transparent;
  background-image: url('./star.svg'), url('./star.svg');
  background-repeat: no-repeat;
  background-size: 60px;
  background-position: left, right;
  font-size: 24px;
  border-top: 3px solid #FFFF00;
  color: #FFFF00;
  text-align: center;
}

.card-button:hover {
  color: white;
  background-image: url('./star-white.svg'), url('./star-white.svg');
}

