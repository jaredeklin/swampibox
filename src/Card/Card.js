import React from 'react';
import './card.css';
import PropTypes from 'prop-types';

export const Card = ({properties, addToFavorites, buttonClass}) => {

  const displayCards = Object.keys(properties)
    .filter(property => property !== "name")
    .map((info, index) => {
      return (
        <li key={index}>
          <span className="card-keys">
          {info}</span>: {properties[info]} 
        </li>)
  })

  return (
    <div className={buttonClass}>
      <h2>{properties.name}</h2>   
        {displayCards}  
      <button onClick={() => addToFavorites(properties)} className='card-button'></button>
    </div>
  ) 
}

Card.propTypes = {
  properties: PropTypes.object,
  addToFavorites: PropTypes.func,
  className: PropTypes.string
}