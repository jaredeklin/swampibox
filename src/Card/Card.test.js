import React from 'react';
import { shallow } from 'enzyme';
import { Card } from './Card';

describe('Card', () => {

  it('should match the snapshot', () => {
    const mockAddToFavorites = jest.fn();
    const mockProps = { 
      capacity: "1",
      class: "repulsorcraft",
      model: "X-34 landspeeder",
      name: "X-34 landspeeder"
    };

    let wrapper = shallow(
      <Card 
        addToFavorites={mockAddToFavorites}
        properties={mockProps}
      />);
    expect(wrapper).toMatchSnapshot();
  });
});