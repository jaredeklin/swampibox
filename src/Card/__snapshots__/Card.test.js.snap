// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    addToFavorites={[Function]}
    properties={
        Object {
            "capacity": "1",
            "class": "repulsorcraft",
            "model": "X-34 landspeeder",
            "name": "X-34 landspeeder",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2>
          X-34 landspeeder
</h2>,
        Array [
          <li>
            <span
                        className="card-keys"
            >
                        capacity
            </span>
            : 
            1
</li>,
          <li>
            <span
                        className="card-keys"
            >
                        class
            </span>
            : 
            repulsorcraft
</li>,
          <li>
            <span
                        className="card-keys"
            >
                        model
            </span>
            : 
            X-34 landspeeder
</li>,
        ],
        <button
          className="card-button"
          onClick={[Function]}
>
          Favorite
</button>,
      ],
      "className": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "X-34 landspeeder",
        },
        "ref": null,
        "rendered": "X-34 landspeeder",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <span
              className="card-keys"
>
              capacity
</span>,
            ": ",
            "1",
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "capacity",
              "className": "card-keys",
            },
            "ref": null,
            "rendered": "capacity",
            "type": "span",
          },
          ": ",
          "1",
        ],
        "type": "li",
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <span
              className="card-keys"
>
              class
</span>,
            ": ",
            "repulsorcraft",
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "class",
              "className": "card-keys",
            },
            "ref": null,
            "rendered": "class",
            "type": "span",
          },
          ": ",
          "repulsorcraft",
        ],
        "type": "li",
      },
      Object {
        "instance": null,
        "key": "2",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <span
              className="card-keys"
>
              model
</span>,
            ": ",
            "X-34 landspeeder",
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "model",
              "className": "card-keys",
            },
            "ref": null,
            "rendered": "model",
            "type": "span",
          },
          ": ",
          "X-34 landspeeder",
        ],
        "type": "li",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Favorite",
          "className": "card-button",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Favorite",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2>
            X-34 landspeeder
</h2>,
          Array [
            <li>
              <span
                            className="card-keys"
              >
                            capacity
              </span>
              : 
              1
</li>,
            <li>
              <span
                            className="card-keys"
              >
                            class
              </span>
              : 
              repulsorcraft
</li>,
            <li>
              <span
                            className="card-keys"
              >
                            model
              </span>
              : 
              X-34 landspeeder
</li>,
          ],
          <button
            className="card-button"
            onClick={[Function]}
>
            Favorite
</button>,
        ],
        "className": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "X-34 landspeeder",
          },
          "ref": null,
          "rendered": "X-34 landspeeder",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span
                className="card-keys"
>
                capacity
</span>,
              ": ",
              "1",
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "capacity",
                "className": "card-keys",
              },
              "ref": null,
              "rendered": "capacity",
              "type": "span",
            },
            ": ",
            "1",
          ],
          "type": "li",
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span
                className="card-keys"
>
                class
</span>,
              ": ",
              "repulsorcraft",
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "class",
                "className": "card-keys",
              },
              "ref": null,
              "rendered": "class",
              "type": "span",
            },
            ": ",
            "repulsorcraft",
          ],
          "type": "li",
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span
                className="card-keys"
>
                model
</span>,
              ": ",
              "X-34 landspeeder",
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "model",
                "className": "card-keys",
              },
              "ref": null,
              "rendered": "model",
              "type": "span",
            },
            ": ",
            "X-34 landspeeder",
          ],
          "type": "li",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Favorite",
            "className": "card-button",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Favorite",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
